test/integration/targets/openssl_dhparam/tasks/main.yml

```diff
@@ -1,88 +1,38 @@
-        size: 768
-        path: '{{ output_dir }}/dh768.pem'
-      register: dhparam_changed
+- name: Run module with backend autodetection
+  openssl_dhparam:
+    path: '{{ output_dir }}/dh_backend_selection.pem'
+    size: 512
 
-    - name: Generate parameters with size option
-      openssl_dhparam:
-        path: '{{ output_dir }}/dh512.pem'
-        size: 512
+- block:
+  - name: Running tests with OpenSSL backend
+    include_tasks: impl.yml
 
-    - name: Don't regenerate parameters with size option and no change
-      openssl_dhparam:
-        path: '{{ output_dir }}/dh512.pem'
-        size: 512
-      register: dhparam_changed_512
+  - include_tasks: ../tests/validate.yml
 
-    - copy:
-        src: '{{ output_dir }}/dh768.pem'
-        remote_src: yes
-        dest: '{{ output_dir }}/dh512.pem'
+  vars:
+    select_crypto_backend: openssl
+  # when: openssl_version.stdout is version('1.0.0', '>=')
 
-    - name: Re-generate if size is different
-      openssl_dhparam:
-        path: '{{ output_dir }}/dh512.pem'
-        size: 512
-      register: dhparam_changed_to_512
+- name: Remove output directory
+  file:
+    path: "{{ output_dir }}"
+    state: absent
 
-    - name: Force re-generate parameters with size option
-      openssl_dhparam:
-        path: '{{ output_dir }}/dh512.pem'
-        size: 512
-        force: yes
-      register: dhparam_changed_force
+- name: Re-create output directory
+  file:
+    path: "{{ output_dir }}"
+    state: directory
 
-    - name: Create broken params
-      copy:
-        dest: "{{ output_dir }}/dhbroken.pem"
-        content: "broken"
-    - name: Regenerate broken params
-      openssl_dhparam:
-        path: '{{ output_dir }}/dhbroken.pem'
-        size: 512
-        force: yes
-      register: output_broken
+- block:
+  - name: Running tests with cryptography backend
+    include_tasks: impl.yml
 
-    - name: Generate params
-      openssl_dhparam:
-        path: '{{ output_dir }}/dh_backup.pem'
-        size: 512
-        backup: yes
-      register: dhparam_backup_1
-    - name: Generate params (idempotent)
-      openssl_dhparam:
-        path: '{{ output_dir }}/dh_backup.pem'
-        size: 512
-        backup: yes
-      register: dhparam_backup_2
-    - name: Generate params (change)
-      openssl_dhparam:
-        path: '{{ output_dir }}/dh_backup.pem'
-        size: 512
-        force: yes
-        backup: yes
-      register: dhparam_backup_3
-    - name: Generate params (remove)
-      openssl_dhparam:
-        path: '{{ output_dir }}/dh_backup.pem'
-        state: absent
-        backup: yes
-      register: dhparam_backup_4
-    - name: Generate params (remove, idempotent)
-      openssl_dhparam:
-        path: '{{ output_dir }}/dh_backup.pem'
-        state: absent
-        backup: yes
-      register: dhparam_backup_5
+  - include_tasks: ../tests/validate.yml
 
-    - import_tasks: ../tests/validate.yml
+  vars:
+    select_crypto_backend: cryptography
```

Comment:
do we have ```auto``` choice anywhere?

Reply:
Yes, the call to `openssl_dhparam` at the top of this file uses the implicit `auto` default (see `Run module with backend autodetection`).

Reply:
ok, thanks for clarifying